File: backend/src/main/scala/microservices/advertising/objects/SaveAdvertisementDeliverySettingsRequest.scala
// 本文件定义 advertising 模块的 SaveAdvertisementDeliverySettingsRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SaveAdvertisementDeliverySettingsRequest(
    placement: String,
    rotationIntervalSeconds: Int,
    playOrder: String,
    startAt: Option[String],
    endAt: Option[String],
    updatedByManagerId: String
)

object SaveAdvertisementDeliverySettingsRequest:
  given sourceEncoder: Encoder[SaveAdvertisementDeliverySettingsRequest] = deriveEncoder
  given sourceDecoder: Decoder[SaveAdvertisementDeliverySettingsRequest] = deriveDecoder

File: frontend/src/microservices/advertising/objects/SaveAdvertisementDeliverySettingsRequest.ts
// 本文件定义 advertising 模块的 `SaveAdvertisementDeliverySettingsRequest`，用于保存投放配置并提供 JSON 编解码。

export type SaveAdvertisementDeliverySettingsRequest = {
  placement: string
  rotationIntervalSeconds: number
  playOrder: string
  startAt: string | null
  endAt: string | null
  updatedByManagerId: string
}

export const saveAdvertisementDeliverySettingsRequestFromJson = (json: string): SaveAdvertisementDeliverySettingsRequest =>
  JSON.parse(json) as SaveAdvertisementDeliverySettingsRequest

export const saveAdvertisementDeliverySettingsRequestToJson = (value: SaveAdvertisementDeliverySettingsRequest): string =>
  JSON.stringify(value)
